controllers/virtualmachine/virtualmachine_controller.go

```diff
@@ -211,15 +211,6 @@ func (r *Reconciler) Reconcile(ctx goctx.Context, req ctrl.Request) (_ ctrl.Resu
 		VM:      vm,
 	}
 
-	// If the VM has a pause reconcile annotation, it is being restored on vCenter. Return here so our reconcile
-	// does not replace the VM being restored on the vCenter inventory.
-	//
-	// Do not requeue the reconcile here since removing the pause annotation will trigger a reconcile anyway.
-	if _, ok := vm.Annotations[vmopv1.PauseAnnotation]; ok {
```

Comment:
Previously this annotation also prevented ReconcileDelete() from being called. Is that something we intended to remove in this MR?

Reply:
Good point. I will check PauseAnnotation when calling ReconcileDelete. But in the case where admin adds PauseEC key and devops deletes the VM soon after, we can't prevent VM getting deleted, right?

Reply:
Hmm. interesting. What is the expected workflow for the case you're describing above, @zyiyi11 ?. Who overrides who?.

From what I see in the implementation, the admin's control to pause reconcile is more like preventing any updates to the VM since it's via the ExtraConfig field set on the VM.  
- Tangential question: Talking about adding the extraConfig fields to the VM, how does the admin set this EC?. I guess this can happen via the VM Class or directly via the OVF?

Earlier the devops could outright pause all reconciles ( create/update/delete) since it was an annotation and the check was here in the controller.

Reply:
> how does the admin set this EC?

@sreyasn VI Admins can set it in the VC UI or via govc as well.

Reply:
On a second thought, I will add check on `PausedVMLabelKey` when triggering ReconcileDelete. So in this case, devops and admins can pause delete reconciles.

Reply:
> But in the case where admin adds PauseEC key and devops deletes the VM soon after, we can't prevent VM getting deleted, right

@zyiyi11 , For this, the only way you could reliably prevent deletion from a devOps user when the EC was added by the Admin later is to check the EC directly in DeleteVirtualMachine(). Since the conditions VirtualMachineReconcileReady=False + PausedReason may not be updated by the time the devOps user tries the delete.

Reply:
> do we allow 'Edit VM settings' to add Advanced Params/ExtraConfig via the VC UI for a VM Service VM?. hmm. never tried it via govc on a VM Service VM, that's probably possible.

@sreyasn Yep, it's doable in the UI once the VM is powered off. And there is also a warning prompt stating that the VM is managed by WCP. For govc, there is no restriction or warning prompt AFAIK.